dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/registry/RegistryClient.java

```diff
@@ -97,7 +98,7 @@ public List<Server> getServerList(NodeType nodeType) {
 
         List<Server> serverList = new ArrayList<>();
         for (Map.Entry<String, String> entry : serverMaps.entrySet()) {
-            HeartBeat heartBeat = HeartBeat.decodeHeartBeat(entry.getValue());
+            HeartBeatModel heartBeat = HeartBeatUtils.decodeHeartBeat(entry.getValue(), nodeType);
```

Comment:
You can make `HeartBeatUtil.decodeHeartBeat` return `AbstractHeartBeat` , then the `HeartBeatModel` can be removed.

Reply:
Worker fields are needed in some places. like host manager, Of course, it can also be obtained by split string of heartbeat ,but it is not graceful.

Reply:
> You can make `HeartBeatUtil.decodeHeartBeat` return `AbstractHeartBeat` , then the `HeartBeatModel` can be removed.

When I designed `WorkerHeartBeat` and `MasterHeartBeat`, I have a dependency cycle problem, so I defined `HeartBeatModel` in module `dolphinscheduler-common` that contains all heartbeatInfo, which also avoids the need to hard code the worker's Heartbeat in the master, For example, the workWaitingTaskCount attribute is used in HostManager.

current design: 
`MasterHeartBeat` in `dolphinscheduler-master`
`WorkerHeartBeat `in `dolphinscheduler-worker`
`AbstractHeartBeat `in `dolphinscheduler-server` (move into `dolphinscheduler-common` is also ok)
`HeartBeatModel `in `dolphinscheduler-common`

ps:
`WorkerHeartBeat `dependency injection `WorkerManagerThread `of dolphinscheduler-worker
`HostWeight `of dolphinscheduler-master use `workWaitingTaskCount `of WorkerHeartBeat 

I don't have any good ideas right now. I need your help.

Reply:
When we get a HeartBeatInfo, we need to know this HeartBeatInfo is represent to worker or master, rather than use a big pojo to contains all fields. If we use a big pojo to represent the HeartBeatInfo, it's hard to know which field is available.

Reply:
First , thank you very much for your patient explanation And I really agree with you.  If `HeartBeatModel ` is deleted, there may be some hard coding in the master, which I cannot accept at present. Therefore, I am ready to give up this PR. I hope that if someone changes it, you can tell me. I want to learn it

Reply:
You don't need to give up this PR, this PR is meaningful. Could you please describe `hard coding in the master`? If you mean the master need to judge the HeartBeatInfo type I think this is needed.

Reply:
At present, I think it's hard coding.

![image](https://user-images.githubusercontent.com/29919212/176172028-a7263ddc-a82f-4d29-ad11-67fd791b1858.png)

Reply:
WorkerHeartBeat in dolphinscheduler-worker. 

PS:WorkerHeartBeat dependency injection WorkerManagerThread of dolphinscheduler-worker

Reply:
OK, I got you....... If so, I suggest we can move the `WorkerHeartBeat` and `MasterHeartBeat` into a `common` module which can be used in api/master/worker.

Reply:
First, we need to reach a consensus that `WorkerHeartBeat` must autowared `WorkerManagerThread` in order to get `WorkerWaitingTaskCount`.

As you said，`WorkerHeartBeat` in `common` module, but `worker` module depend` common` module and `WorkerManagerThread` in `worker` module.
![image](https://user-images.githubusercontent.com/29919212/176203192-8b57a207-1a94-4f05-b94b-0724f7914b71.png)

Reply:
WorkHeartBeat is just a pojo, it should not depend anything, if I have any misunderstand, please let me know.

Reply:
The worker needs to update `workWaitingTaskCount `to report the heartbeat information.

Reply:
Yes, the worker will schedule a WorkerHeartbeatTask to update the info in WorkHeartBeat, there seems no problem.

Reply:
I have some questions to confirm.
-  1. Do we need to define a `WorkerHeartbeatTask` ?
- 2. What do you mean `WorkerHeartbeatTask`  autowared `WorkerManagerThread` ?
- 3. Which module is `WorkerHeartbeatTask` in ?